### PR TITLE
Explain the customCheProperties field

### PR DESCRIPTION
**Pull request by tomgeorge**
### What does this PR do?

Explains the new `overrideCheProperties` field in the `CheCluster` Custom Resource and shows an example CR that has an additional property.

### What issues does this PR fix or reference?
eclipse/che#14635

### CODE REVIEW

**Comment by tomgeorge** [comment]
I think we are going to change the field name in the `CheCluster` cr, so these docs will need to change.

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc:19]: I would replace:

*main properties __for the installation__*

by 

*main properties __for the Che server__*

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc]:
```suggestion
To add custom properties to the Che server, such as environment variables that are not automatically generated in the `che` `configMap` by the operator, or to override automatically generated properties, the `CheCluster` Custom Resource has a `customCheProperties` field, that expects a map.  
```

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc]:
```suggestion
For example, to add a property named `FOO` with value `BAR`, add an entry in `customCheProperties`:
```

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc]:
```suggestion
Previous versions of the Che operator had a configMap named `custom` to fulfill this role.  If the Che operator finds a `configMap` with the name `custom`, it will add the data it contains into the `customCheProperties` field, redeploy Che, and delete the `custom` `configMap`. 
```

**Review by davidfestal — CHANGES_REQUESTED** [review]
Please look at my comments

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc]:
```suggestion
Che installed using the Operator provides the user with an automatically generated `configMap` called `che`.
```

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (rkratky) [inline src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc]:
```suggestion
The `che` `configMap` contains the main properties for the Che server and is in sync with the information stored in the CheCluster Custom Resource file. User modifications of the `che` `configMap` after installing Che using the Operator are automatically overwritten by values that the Operator obtains from the `CheCluster` Custom Resource. 
```

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (rkratky) [reply src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc]: Building on David's suggestion:

```suggestion
To add custom properties to the Che server, such as environment variables that are not automatically generated in the `che` `configMap` by the Operator, or to override automatically generated properties, the `CheCluster` Custom Resource has a `customCheProperties` field, which expects a map.  
```

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (rkratky) [reply src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc]: Building on David's suggestion:

```suggestion
Previous versions of the Che Operator had a configMap named `custom` to fulfill this role.  If the Che Operator finds a `configMap` with the name `custom`, it adds the data it contains into the `customCheProperties` field, redeploys Che, and deletes the `custom` `configMap`. 
```

**Review by rkratky — CHANGES_REQUESTED** [review]

**Comment by tomgeorge** [comment]
@rkratky @davidfestal I made the changes that you suggested.

File: src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc
Comment (davidfestal) [inline src/main/pages/che-7/installation-guide/con_che-configmaps-and-their-behavior.adoc]: I'm not sure we need to add all the other fields in the other sections...
Maybe ... would be sufficient no ?

**Review by davidfestal — CHANGES_REQUESTED** [review]

**Comment by tomgeorge** [comment]
@rkratky we are ready with https://github.com/eclipse/che-operator/pull/87.  Once this is approved we will merge

**Review by davidfestal — APPROVED** [review]
All is OK on my side.
Let's just wait for docs team approval before merging the main PR.

**Review by nickboldt — CHANGES_REQUESTED** [review]
Instead of `FOO: BAR` why not include some (or all) of the image and pull_policy references in che.properties, so we have an example of WHY you'd use this override mechanism?

eg. for productization and using chectl to deploy CRW:

`CHE_SERVER_SECURE__EXPOSER_JWTPROXY_IMAGE: 'registry.redhat.io/codeready-workspace/jwtproxy-rhel8:2.0'`

and for airgap:
`CHE_FACTORY_DEFAULT__PLUGINS: 'myquay.mycorp.com/che-machine-exec-plugin/7.2.7'`
`CHE_WORKSPACE_SIDECAR_IMAGE__PULL__POLICY: 'Never'`

I would also link to https://github.com/eclipse/che/blob/master/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties and explicitly explain how those properties map to fields in a yaml file: 

* uppercase the parameter
* replace `.` with `_` and `_` with `__`

Thus:

`che.workspace.plugin_broker.pull_policy` -> `CHE_WORKSPACE_PLUGIN__BROKER_PULL__POLICY`

**Comment by davidfestal** [comment]
> I would also link to https://github.com/eclipse/che/blob/master/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties and explicitly explain how those properties map to fields in a yaml file:
> 
>     * uppercase the parameter
> 
>     * replace `.` with `_` and `_` with `__`
> 
> 
> Thus:
> 
> `che.workspace.plugin_broker.pull_policy` -> `CHE_WORKSPACE_PLUGIN__BROKER_PULL__POLICY`

Seems to me that this is more related to a different docs issue:

https://github.com/eclipse/che/issues/13886#issuecomment-534171945

I'd rather not block the merge of PR https://github.com/eclipse/che/issues/14635 for this.

**Comment by davidfestal** [comment]
> eg. for productization and using chectl to deploy CRW:
> 
> `CHE_SERVER_SECURE__EXPOSER_JWTPROXY_IMAGE: 'registry.redhat.io/codeready-workspace/jwtproxy-rhel8:2.0'`
> 
> and for airgap:
> `CHE_FACTORY_DEFAULT__PLUGINS: 'myquay.mycorp.com/che-machine-exec-plugin/7.2.7'`
> `CHE_WORKSPACE_SIDECAR_IMAGE__PULL__POLICY: 'Never'`

Maybe I would keep the air-gap aspect for this other Che docs PR : https://github.com/eclipse/che/issues/14088#issuecomment-534174598

**Review by nickboldt — DISMISSED** [review]
Ok, @davidfestal those other issues can cover the doc issues. 

But I'd still rather see a real(ish) example than `FOO:BAR` like 

`CHE_SERVER_SECURE__EXPOSER_JWTPROXY_IMAGE: 'registry.redhat.io/codeready-workspace/jwtproxy-rhel8:2.0'`

**Review by rkratky — APPROVED** [review]

**Review by themr0c — APPROVED** [review]

**Comment by davidfestal** [comment]
@themr0c why did you merge this already ? Shouldn't we wait for the corresponding main PR to be merged to master and, even more, released to production with the 7.3.0 release ?

**Comment by themr0c** [comment]
How should I have known?

**Comment by davidfestal** [comment]
> How should I have known?

I thought it was the usual way: only merge a doc change when the related product change is released.
But I should have been clearer in the PR description it seems.

**Comment by themr0c** [comment]
Yes please prepend "[WIP]" to the title of your PR when it should not me merged, and remove it once it can be merged.

It's first time we are ahead of the product, documentation is usually lagging behind, so I didn't even thought about verifying the feature is present!

**Pull request merged**
